docs/en/resource/supplemental.md

```diff
@@ -0,0 +1,351 @@
+
+The transform is transparent and occurs in the middleware before data is returned to the widget. When performing queries on the resource, a reverse transform is used to convert the widget values back to the source values provided.
+
+```ts
+// resource data
+const resourceData = [{ firstname: 'joe', lastname: 'bloggs' }, { firstname: 'jane', lastname: 'doe' }];
+
+// with a transform of
+const transform = {
+	value: ['firstname']
+};
+
+// would create
+const transformedData = [{ value: 'joe' }, { value: 'jane' }];
+```
+
+Transforms can also be used to join source keys together to create composite fields. For example, with the same resource data as above
+
+```ts
+// with a transform of
+const transform = {
+	value: ['firstname', 'lastname']
+};
+
+// would create
+const transformedData = [{ value: 'joe bloggs' }, { value: 'jane doe' }];
+```
+
+### using a typed transform
+
+```tsx
+import { create } from '@dojo/framework/core/vdom';
+import { createDataMiddleware } from '@dojo/framework/core/middleware/data';
+
+// create the middleware with a generic
+const data = createDataMiddleware<{ value: string }>();
+const factory = create({ data });
+
+export const DataAwareWidget = factory(function ({
+	middleware: { data }
+}) {
+	const { getOrRead, getOptions } = data();
+	const data = getOrRead(getOptions());
+
+	return (
+		<virtual>
+			{ data.map(item => <span>{item.value}</span>) }
+		</virtual>;
+	)
+}
+
+// in render function elsewhere
+<List resource={resource} transform={{ value: ['firstname', 'lastname']}}>
+// if a key other than `value` was passed in the transform or no transform at all
+// was passed, a type error would occur
+```
+
+## API
+
+The data middleware API provides the widget with access to the resource data.
+
+### get options
+
+`getOptions` returns the current `Options` object. The result of this function should be passed to each of the `get` api functions. It is important to use `getOptions` rather than constructing a new `Options` object in order to ensure that shared resources etc work as expected.
+
+### set options
+
+`setOptions` is used to update the current `Options` on the resource wrapper. This will reflect in any other widget using the shared resource.
+
+```ts
+const { getOptions, setOptions } = data();
+
+setOptions({
+	...getOptions(),
+	pageNumber: 1
+});
+```
+
+### get
+
+The `get` function takes an `Options` object and returns data as requested if it is already available on the resource. It will not perform a `read`.
+
+```ts
+const { getOptions, get } = data();
+
+const data = get(getOptions());
+```
+
+### getorread
+
+The `getOrRead` function takes an `Options` object and returns the data requested. If it is not already availablee, it will perform a `read`. When the data is available the widget will be invalidated so it can be used in a reactive manner.
```

Comment:
```suggestion
The `getOrRead` function takes an `Options` object and returns the requested data. If the data is not already available, it will perform a `read`. Once the data is available, the widget will be invalidated in a reactive manner.
```